content/_code-samples/freeze/py/unfreeze_token.py

```diff
@@ -0,0 +1,49 @@
+from xrpl.clients import JsonRpcClient
+from xrpl.models import IssuedCurrencyAmount, TrustSet
+from xrpl.transaction import (safe_sign_and_autofill_transaction,
+                              send_reliable_submission)
+from xrpl.models import TrustSetFlag
+from xrpl.utils import str_to_hex
+from xrpl.wallet import generate_faucet_wallet
+
+client = JsonRpcClient("https://s.altnet.rippletest.net:51234") # connect to testnet
+
+token_name = "USD"
+
+# Amount a trustline can handle
+value = "0"
+
+# Address to unfreeze trustline
+target_addr = generate_faucet_wallet(client=client).classic_address
+
+# Sender wallet
+sender_wallet = generate_faucet_wallet(client=client)
+
+print("Successfully generated test wallets")
+
+# Build trustline freeze transaction
+trustset = TrustSet(account=sender_wallet.classic_address, limit_amount=IssuedCurrencyAmount(
+    currency=token_name,
+    issuer=target_addr,
+    value = value 
+),
+flags=TrustSetFlag.TF_CLEAR_FREEZE)
+
+print("prepared and submitting TF_CLEAR_FREEZE transaction")
+
+# Sign and Submit transaction
+stxn = safe_sign_and_autofill_transaction(trustset, sender_wallet, client)
+stxn_response = send_reliable_submission(stxn, client)
+
+# Parse response for result
+stxn_result = stxn_response.result
+
+# Print result and transaction hash
+if stxn_result["meta"]["TransactionResult"] == "tesSUCCESS":
+  print(f'Successfully enabled no freeze for {sender_wallet.classic_address}')
+if stxn_result["meta"]["TransactionResult"] == "tecNoLineRedundant":
+  print("This was used on an account which didn't have a trustline yet. To try this out, modify `target_addr` to point to an account with a frozen trustline, and make sure the currency code matches.")
+else:  
+  print(stxn_result["meta"]["TransactionResult"])
```

Comment:
This actually has to be in a try/catch block or else you get an error like this when `tecNoLineRedundant` happens:

```
python3 unfreeze_token.py
Successfully generated test wallets
prepared and submitting TF_CLEAR_FREEZE transaction
Traceback (most recent call last):
  File "/Users/jacksonmills/Documents/obiajulu-2/xrpl-dev-portal/content/_code-samples/freeze/py/unfreeze_token.py", line 35, in <module>
    stxn_response = send_reliable_submission(stxn, client)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/xrpl/transaction/reliable_submission.py", line 29, in send_reliable_submission
    return asyncio.run(async_send_reliable_submission(transaction, client))
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/asyncio/base_events.py", line 641, in run_until_complete
    return future.result()
  File "/Library/Frameworks/Python.framework/Versions/3.10/lib/python3.10/site-packages/xrpl/asyncio/transaction/reliable_submission.py", line 86, in send_reliable_submission
    raise XRPLReliableSubmissionException(
xrpl.asyncio.transaction.reliable_submission.XRPLReliableSubmissionException: Transaction failed, tecNO_LINE_REDUNDANT: Can't set non-existent line to default.
```

Reply:
<img width="461" alt="image" src="https://user-images.githubusercontent.com/47371105/203023792-40ee2903-c92a-4b21-9af9-0fc69582c5a8.png">
This is what I get when it returns a `tecNoLineRedundant`

Reply:
I don't know if its an SDK version issue, I'm using 1.7.0

Reply:
I pushed an update, try and see if this happens again

Reply:
I made some changes, try and see if this happens again

Reply:
Good call - I had xrpl-py version 1.3, but once I upgraded to 1.7 the code worked as you see in your screenshot :)